Sync v2-6-stable with v2-6-test to release 2.6.2

eladkal opened the pull request:
Time for `2.6.2rc1`!

o-nikolas committed:
Export SUSPENDED_PROVIDERS_FOLDERS for breeze testing commands (#30780)

Export the SUSPENDED_PROVIDERS_FOLDERS env var in breeze directly
instead of in Airflow CI workflows. This will fix the issue for users
executing `breeze testing ...` commands locally.

(cherry picked from commit b04894e5125e682d9ad043602d4babe5b0b39b0c)

potiuk committed:
Speed up test collection (#30801)

Test collection had default setting for parallel test types because
TEST_TYPES variable had not been renamed to PARALLEL_TEST_TYPES
Also test collection can be run in Wait for CI images job which
should save around a minute for setting up Breeze and pulling the
images.

This should speed up pytest collection test by around 1 and half minute

(cherry picked from commit 93a5422c5677a42b3329c329d65ff2b38b1348c2)

potiuk committed:
Add instructions on how to avoid accidental airflow upgrade/downgrade (#30813)

Some of our users raised issues that when extending the image, airflow
suddenly started reporting problem with database versions and migration
not aplied or out-of-sync. This almost always turns out to be a
dependency conflict, that leads to automated downgrate or upgrade of
installed airflow version. This is - obviously - undesired (you should
be upgrading airflow consciously rather than accidentally). However
there is no way to do it implicitly - `pip` might decide to upgrade or
downgrade airflow as it sees fit. From the point of view - airflow is
just one of the packages and has no special meaning.

The only way to "keep" airflow version is to specify it together with
other requirements, pinned to the specific version. This PR updates
our examples to do this and explains why airflow is added there.

There is - of course - another risk that the user will forget to
update the version of airflow when they upgrade, however, sinc this
is explicit action performed during image extension, it is much easier
to diagnose and notice. We also warn the users that they should upgrade
when airflow is upgraded.

(cherry picked from commit bf6ebe990cbcc290b5b9cd1060573f3cf44782ad)

potiuk committed:
Optimize away pytest collection steps (#30824)

The Pytest collection steps are only needed if there are any tests
about to be run. There are cases where we build CI images but
we do not expect to run any tests (for doc-only changes). This
should save about 3 minutes of build time.

Also ARM pytest collection should only be executed once for
every build after all "regular" tests passed - there is no need
to run them for different backends/versions. It is enough
to run them for default single backend version (Postgres).

(cherry picked from commit 557716aa0215bab9b084e926fe09b779f046ac85)

potiuk committed:
Only run mssql tests on self-hosted runners (#30831)

The MSSQL tests are rather flaky on public runners. They also
almost never bring any extra value when fail usually those are
flaky behaviour of MSSQL database for those runners due to
resource constraints.

This PR changes MSSQL tests to be only run for self-hosted runners,
which means that they will be run for PRs of committers and in
canary builds.

This should be enough to find out potentially mssql-only breaking
changes and fix them - without disturbing regular contributors.

(cherry picked from commit 6e80bcd463a701b30ce42f4499ea45d4c4e47972)

raboof committed:
Add broad-strokes 'security model' (#30843)

Add a broad-strokes description of the security expectations
operator should expect. This will get included into
https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/security/index.html

I'm no Airflow expert so help definitely welcome.

I would like this section to be a bit more prominent rather than hidden away
under 'Administration and Deployment', but it looks like the structure was
carefully considered in https://github.com/apache/airflow/pull/27235 so this
is probably fine.

Eventually we could point the 'Security' link on the main pages
like https://airflow.apache.org/ to this page.

(cherry picked from commit f20c08a77c9daefe54c14c5013fcf53d84b5744d)

potiuk committed:
Speed up verification and instalation of providers in CI (#30839)

For PRs that are just changing selected providers, there is no
need to install and verify all providers, just those affected
should be installed (and full verification should only be done
when all providers are being built - so generally speaking, when
Core PRs are run or when canary builds are run.

This PR distinguishes those two cases:

* installing and verification when all providers are selected
* installing only affected providers when only subset of them are
  affected

Also a bug was found where core file changes did not trigger provider
docs build during implementation. This has been fixed

(cherry picked from commit 3af22f5bedad370ccb35ae887dff6daf995fd90c)

huymq1710 committed:
Fix DAG.access_control can't sync when clean access_control (#30340)

* Reset permission if `access_control` is empty

* Check `resource` before call `_revoke_all_stale_permissions`

* Fix static checks

(cherry picked from commit 2c0c8b8bfb5287e10dc40b73f326bbf9a0437bb1)

ying-w committed:
Update callbacks documentation (errors and context) (#31116)

(cherry picked from commit a2a9867ad1fb94ecde31f99d068824396e5d407b)

dstandish committed:
Use only one line for tmp dir log (#31170)

(cherry picked from commit b4ba6b7cb1c232c9956e0a969f015b2ce5964d15)

dstandish committed:
Executor events are not always "exited" here (#30859)

(cherry picked from commit f39ccad9d8fa239bdbeb7e88afdedea5e8fddf76)

Adaverse committed:
Added spinner activity while the logs load (#31165)

* spinner added

* fixed static checks

* isFetching

* isFetching replaced with isLoading

(cherry picked from commit 584a9f5dae5b29a1968fbdbc9b1edd01ae2be4d2)

potiuk committed:
Add description on security issue handling in Airflow (#31160)

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>
Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>
Co-authored-by: Vincent <97131062+vincbeck@users.noreply.github.com>
Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>
(cherry picked from commit 60536152b2466dcfab6a7323ae58beb0867caad1)

mattusifer committed:
Validate connection IDs (#31140)

* Validate connection_ids

(cherry picked from commit 5cb8ef80a0bd84651fb660c552563766d8ec0ea1)

jedcunningham committed:
Fix `url_for_asset` fallback and 404 on DAG Audit Log (#31233)

There was a 404 on the DAG Audit Log for bootstrap toggle css, which as
far as I can tell, we aren't even trying to use. So we will no longer
try and add it to the page.

This highlighted the fact that if we don't have an asset in the
manifest, `url_for_asset` would just leave off the filename completely.
This resulted in the href being just `/static/`, with no filename as a
breadcrumb in the eventual 404. Now we will just use the filename as-is
so the 404 is more meaningful.

(cherry picked from commit 2b4ad8a18c7c3975a30afe86030b2db9f8daf6f6)

yenchenLiu committed:
Fix task.branch error caused by incorrect initial parameter (#31265)

(cherry picked from commit d59b0533e18c7cf0ff17f8af50731d700a2e4b4d)

VVildVVolf committed:
Upgrade FAB to 4.3.1 (#31203)

(cherry picked from commit 1133035f7912fb2d2612c7cee5017ebf01f8ec9d)

uranusjr committed:
Make BaseJobRunner a generic on the job class (#31287)

(cherry picked from commit 2b6462b5a44ea4502c8f4d1b3d7bc9814a65c937)

potiuk committed:
Use keyword paramaters for migration methods for mssql (#31309)

The new alembic (1.11.0) requires parameters to be keyword
parameters in methods used in migrations. Part of the problem has
been fixed in #31306 and #31302 but still some migrations that are
specifically run in case of mssql need to use migration parameters.

(cherry picked from commit 54790274c2edf39c06f494a900bc85523349a77f)

pankajkoti committed:
Rephrase comment in setup.py (#31312)

Following up on my late comment in PR #31309, this is a try to rephrase
the comment in `setup.py` which indicates how to trigger CI build to use
"upgrade to newer dependencies".

(cherry picked from commit 492acc274b278728069da246c88162f938cb0fb0)

potiuk committed:
Convert dask upper-binding into exclusion (#31329)

As https://github.com/dask/dask/issues/10279 has been solved and
Dask tests should no longer generate a PATH side-effect, we
should turn the upper-binding into exclusion, to make sure that
we upgrade to newer Dask version when it is released.

(cherry picked from commit 45f1fda2fac8bf5455a1f444b98b58d38ea2db63)

hussein-awala committed:
Add an example for sql_alchemy_connect_args conf (#31332)

(cherry picked from commit a018f0f5dc9114d06a81efd42ac3f745260507de)

arjunanan6 committed:
Add fullname to owner on logging (#30185)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Malthe Borch <mborch@gmail.com>
Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit a03f6ccb153f9b95f624d5bc3346f315ca3f0211)

lwyszomi committed:
Update SDKs for google provider package (#30067)

* Update SDK versions for Google provider

* Adjust google ads operators to v12

Changes:
- fix tests/system/providers/google/cloud/bigquery/example_bigquery_queries.py
- fix tests/system/providers/google/cloud/bigquery/example_bigquery_queries_async.py

* Fix GCS system tests

* Fix CloudBuild unit test

* Update BigTable operators to accomodate for new dependencies.

* Fix Cloud Tasks System tests

Tasks dag was quite flaky without the retry option in the run_task step,
but it's consistently green with the option set.

We also add a GCP_APP_ENGINE_LOCATION env variable since this depends on
the used GCP Project App Engine's location

* Add setup docstring to Tasks system tests.

* Update Vision operators to accommodate new dependencies.

Changes:
- fix methods for CloudVisionHook
- fix Vision Operators
- fix tests/providers/google/cloud/hooks/test_vision.py
- fix tests/providers/google/cloud/operators/test_vision.py
- fix tests/system/providers/google/cloud/vision/example_vision_annotate_image.py
- fix tests/system/providers/google/cloud/vision/example_vision_autogenerated.py
- fix tests/system/providers/google/cloud/vision/example_vision_explicit.py

* Update SpeechToText operators to accommodate new dependencies.

Changes:
- fix synthesize_speech method for CloudTextToSpeechHook
- fix CloudSpeechToTextRecognizeSpeechOperator
- fix tests/providers/google/cloud/operators/test_speech_to_text.py
- fix tests/providers/google/cloud/hooks/test_text_to_speech.py
- fix tests/providers/google/cloud/hooks/test_speech_to_text.py

* Update Translate Speech operators to accommodate new dependencies.

Changes:
- fix synthesize_speech method for CloudTextToSpeechHook
- fix CloudTranslateSpeechOperator
- tests/providers/google/cloud/operators/test_translate_speech.py

* Update VideoIntelligence operators to accommodate new dependencies.

Changes:
- fix annotate_video method for CloudVideoIntelligenceHook
- fix VideoIntelligence Operators
- fix tests/providers/google/cloud/hooks/test_video_intelligence.py
- fix tests/providers/google/cloud/operators/test_video_intelligence.py

* Update Compute Engine operators to accomodate for new dependencies.

Changes:
- added wait_for_operation_complete() method to check the execution flow
- added new attribute cmd_timeout for ComputeEngineSSHHook

* Fix Stackdriver system test

This test has not worked because of slack channel and credentials not
being setup. We now test the same operators by creating notification
channels and policy alerts against pubsub topics, which don't need to
exist before the test is ran, making the test self-contained.

* Update Natural Language operators to accommodate new dependencies.

Changes:
- fix airflow/providers/google/cloud/operators/natural_language.py
- fix airflow/providers/google/cloud/hooks/natural_language.py
- fix tests/providers/google/cloud/hooks/test_natural_language.py
- fix tests/providers/google/cloud/operators/test_natural_language.py
- fix tests/system/providers/google/cloud/natural_language/example_natural_language.py

* Update Composer system tests.

Fix environment id to contain underscores.

* Update AutoML operators to accommodate new dependencies.

Changes:
- add timeout parameter to all long-running operations for operators
- fix tests/system/providers/google/cloud/automl/example_automl_dataset.py
- fix tests/system/providers/google/cloud/automl/example_automl_model.py
- fix tests/system/providers/google/cloud/automl/example_automl_nl_text_extraction.py
- fix tests/system/providers/google/cloud/automl/example_automl_vision_classification.py

* Fix Cloud SQL delete operator

For some delete instance operations, the operation stops being available ~9 seconds after completion, so we need a shorter sleep time to make sure we don'tmiss the DONE status.

* Update VertexAI operators to accommodate new dependencies.

* Add SQL to Sheets Test instructions

* Update Dataproc Metastore operators to accommodate new dependencies.

* Update Dataproc operators to accommodate new dependencies.

* Update Dataflow sys tests to new sdk

* Update Dataproc on gke operators to accommodate new dependencies.

* Update MLEngine operators to accomodate new dependencies.

Changes:
- update train model that is used for prediction
- update version and runner for ApacheBeam in utils for MLEngine
- update connection inside async hook

* Update Dataprep operators to accommodate new dependencies.

Changes:
- fix tests/system/providers/google/cloud/dataprep/example_dataprep.py

* Add Dataflow Go system test

* Update providers.yaml for google

* fixup! Update providers.yaml for google

* Google SDK Fixes after rebase

- Secret Manager was missing updating to v2, now expects a request dict
- Compute ssh had a bug when no cmd_timeout was passed
- Cloud Build tests were improved/refactored in community, so deleting
old ones
- googleapiclient.errors.HttpError was incorrectly used in our tests, it
it didn´t matter before but a change in the class makes HttpError()
raise an error in initialization the way we were using it before
- fix static checks
* Fix Google providers type errors

---------

Co-authored-by: Lukasz Wyszomirski <wyszomirski@google.com>
Co-authored-by: Maksim Moiseenkov <maksim_moiseenkov@epam.com>
Co-authored-by: Eugene Kostieiev <kosteev@google.com>
Co-authored-by: Augusto Hidalgo <augustoh@google.com>
Co-authored-by: Beata Kossakowska <bkossakowska@google.com>
Co-authored-by: Ulada Zakharava <uladaz@google.com>
Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
(cherry picked from commit 28d1bf8dfc03787cc2b4373bbaa27527bdd91846)

Commit:
Fix ExternalTaskSensor to work correctly with task groups (#30742)

* Fix ExternalTaskSensor to work correctly with task groups that have mapped tasks

* Add tests for ExternalTaskSensor with task group that have mapped tasks

* Fix working for multiple runs

* Use tuple_in_condition instead of tuple_

---------

Co-authored-by: Zhyhimont Dmitry <dzhigimont@gmail.com>
(cherry picked from commit 3c30e54de3b8a6fe793b0ff1ed8225562779d96c)

pankajastro committed:
Make connection id validation consistent across interface (#31282)

* Make connection id validation consistent across interface

recently, we added the connection id validation
if the user creates a connection from UI but
this improvement is missing in CLI and API.
This PR make sure that connection id validation is consistent
across interfaces like CLI, API, and UI.

(cherry picked from commit 7bf53395f2802b9952b0e90ffe548f70c50907ad)

Owen-CH-Leung committed:
Update the output of airflow info command in the doc (#31336)

* Update logging-tasks.rst

Update the output of airflow info command

* Update logging-tasks.rst

* Update logging-tasks.rst

* Fix failing static check

(cherry picked from commit fc4f37b105ca0f03de7cc49ab4f00751287ae145)

rohan472000 committed:
Added metavar and choices to --state flag in airflow dags list-jobs CLI for suggesting valid state arguments. (#31308)

(cherry picked from commit 8e296a09fc5c49188a129356caca8c3ea5eee000)

uranusjr committed:
Skip auto-document things with :meta private: (#31380)

(cherry picked from commit f81402e3fdaf58f83a14fe5ec5a8a782dfb13b99)

potiuk committed:
Check Logging Handlers and Secret Backends for recursion (#31385)

Logging Handlers and secret backends have the potential of triggering
Import/Attribute error due to circular imports. This check will
attempt to import all those classes from our providers via
airflow_local_settings, which should detect the case where such
circular import condition is generated.

(cherry picked from commit 33709f07afc337ddf429afcc74707a403fcf3513)

jedcunningham committed:
Fix error handling when pre-importing modules in DAGs (#31401)

(cherry picked from commit 24a94bbb603c5308a2e8817dc356492287f7b174)

potiuk committed:
Add libgeos-dev to dev dependencies in CI image (#31437)

When running some Mlengine tests, pytest (when running assert rewrite)
attempts to load libgeos-dev library, which breaks collection of some
Google Provider tests on ARM image (it is used by Shapely library
imported by recent BigQuery library)

This PR Adds the library as preinstalled in the CI image.

(cherry picked from commit 9137740b2e8937aaf65c7c86dd11b096d028e7d2)

jscheffl committed:
Fix dropdown default and adjust tutorial to use 42 as default for proof (#31400)

(cherry picked from commit 58aab1118a95ef63ba00784760fd13730dd46501)

potiuk committed:
Add description in a few more places about adding airflow to pip install (#31448)

It is important to keep airflow as a dependency with pinned version if
you are adding new dependencies. This PR adds it in case of
docker-compose documentation and also improves examples in our
images, by using AIRFLOW_VERSION variable defined in the images. This
way users can write future-proof ways of installind dependencies basing
on our examples.

Follow-up after #30813

(cherry picked from commit d722fba54e844b83025a7189b28371fba7e1af88)

tadeha committed:
Add missing BashOperator import to documentation example (#31436)

* Update notifications.rst with importing BashOperator in the example.

* Sort imports

* Update docs/apache-airflow/howto/notifications.rst

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>
(cherry picked from commit 1ac35e710afc6cf5ea4466714b18efacdc44e1f7)

potiuk committed:
Add Python 3.11 support (#27264)

Add Python 3.11 support

Python 3.11 has been released as scheduled on October 25, 2022 and
finally, after all dependencies got upgraded - we can support it.

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit c5597d1fabe5d8f3a170885f6640344d93bf64bf)

potiuk committed:
Fix version support information (#31468)

The Python 3.11 change updated version support for stable version.

This changes updates this and also fixes a few small requirement
miscommunications.

(cherry picked from commit 511d25d0fc9b7db033ffa5a0ad2ef9ea4bdaa305)

hussein-awala committed:
Ensure min backoff in base sensor is at least 1 (#31412)

(cherry picked from commit a98621f4facabc207b4d6b6968e6863845e1f90f)

potiuk committed:
Fix table formatting in docker build documentation (#31472)

(cherry picked from commit df9031742edce7758aed111f4cb2332bcf91f924)

jedcunningham committed:
Remove note about triggerer being 3.7+ only (#31483)

3.7 is now the lowest Python version Airflow supports, so no need to
explicitly call out that the triggerer requires 3.7+.

(cherry picked from commit 89dc1cd17d3aaca88da2be00542b946345685b3e)

uranusjr committed:
Fix typing in external task triggers (#31490)

(cherry picked from commit b3536217cd80bda8b56068a2efb3fa6979d17b3f)

luos-fc committed:
Use kube_client over default CoreV1Api for deleting pods (#31477)

(cherry picked from commit adf0cae48ad4e87612c00fe9facffca9b5728e7d)

ephraimbuddy committed:
Cascade update of taskinstance to TaskMap table (#31445)

(cherry picked from commit f6bb4746efbc6a94fa17b6c77b31d9fb17305ffc)

ephraimbuddy committed:
Remove dependency already registered for this task warning (#31502)

(cherry picked from commit abcfc6f76def56c56c0bb15423eab00a837c099a)

AutomationDev85 committed:
Save scheduler execution time during search for queued dag_runs (#30699)

* Function returns list of dagruns and not query

* Changed pytests

* Changed all to _start_queued_dagruns

* Added comment and fixed tests

* Fixed typo

(cherry picked from commit 0fd42ff015be02d1a6a6c2e1a080f8267194b3a5)

hussein-awala committed:
fix airflow users delete CLI command (#31539)

* fix airflow users delete CLI command

* replace setting to empty list by clear method

(cherry picked from commit 3ec66bb7cc686d060ff728bb6bf4d4e70e387ae3)

potiuk committed:
Workaround mypy confusing default with necessary arguments (#31562)

Mypy heuristics gets the typing wrong in those cases, so we
need to ignore them for now

(cherry picked from commit a6bef887298dbbe107efbc0e4634998f1b60f0e7)

Adaverse committed:
Resolving issue where Grid won't un-collapse when Details is collapsed (#31561)

* spinner added

* fixed static checks

* isFetching

* isFetching replaced with isLoading

* Update CONTRIBUTING.rst

* - fix where collapsed Grid won't show up when Details is uncollapsed
- removing clearSelection when Details is collapsed
- making experience uniform when collapsing or uncollapsing Grid or Detaials view

* resize oberserver re-registering when Grid is un-collapsed

* reverting 28px to show atleast one (latest) dag run instance

(cherry picked from commit 6f86b6cd070097dafca196841c82de91faa882f4)

arjunanan6 committed:
Fix sorting of tags (#31553)

* Fix sorting of tags

* Change sorting in views.py

* Add ordering into query

(cherry picked from commit 24e52f92bd9305bf534c411f9455460060515ea7)

pankajkoti committed:
Include triggerer health status in Airflow /health endpoint (#31529)

PR https://github.com/apache/airflow/pull/27755 introduced sending triggerer
health status in the `/api/v1/health` endpoint and also updated relevant
docs but we've the primary `/health` too which is missing this information.
The PR addresses this missing status report for triggerer health in the
`/health` endpoint. It also attempts to deduplicate the code between those
endpoints so that in future we need to make necessary changes in only one
place and at the same time ensure that change made in one endpoint is not
missed for the other endpoint serving the same purpose and thus ensuring
consistency in the responses.

fixes: #31522
(cherry picked from commit f048aba47e079e0c81417170a5ac582ed00595c4)

iJanki-gr committed:
Remove typo in docs serializers.rst (#31507)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit c930016f1c0e99e97a4392a479420198ef2b4e14)

Adaverse committed:
Updating UI screenshots in docs (#31577)

(cherry picked from commit 38fb4329fd3c0663d8ff79d359e9824ee2a082f5)

pierrejeambrun committed:
Add TriggererStatus to OpenAPI spec (#31579)

* Add TriggererStatus to OpenAPI spec to health endpoint

* Update spec

* Update description

(cherry picked from commit fc36d002fccaad65718f42994747600c14f10263)

hussein-awala committed:
Add the missing `map_index` to the xcom key when skipping downstream tasks (#31541)

* Add the missing map_index to the xcom key when skiping downstream tasks

* fix subdag operator tests

* Update tests/operators/test_subdag_operator.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit e2da3151d49dae636cb6901f3d3e124a49cbf514)

AutomationDev85 committed:
Optimize scheduler by skipping "non-scheduable" DAGs (#30706)

Co-authored-by: Jens Scheffler <jens.scheffler@de.bosch.com>
(cherry picked from commit ec18db170745a8b1df0bb75569cd22e69892b3e2)

utkarsharma2 committed:
Hide old alert box when testing conn with different value (#31606)

(cherry picked from commit efe8473385426bf8c1e23a845f1ba26482843855)

hussein-awala committed:
Fix overriding `default_args` in nested task groups (#31608)

* add unit tests for default_args overriding in task group

Signed-off-by: Hussein Awala <hussein@awala.fr>

* fix overriding default args in nested task groups

Signed-off-by: Hussein Awala <hussein@awala.fr>

* Update airflow/utils/task_group.py

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

---------

Signed-off-by: Hussein Awala <hussein@awala.fr>
Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>
(cherry picked from commit 9e8627faa71e9d2047816b291061c28585809508)

dstandish committed:
Remove found_descendents param from get_flat_relative_ids (#31559)

By the looks of it, this param is unused.  Since the class is designated private, it is permissable to remove it.

(cherry picked from commit 0cbc0dc6d8bbebce5e2f039a70eb48bf3ee7a267)

Adaverse committed:
fixing broken links in openapi/v1.yaml (#31619)

* fixing broken links in openapi/v1.yaml

* fixed static check

(cherry picked from commit 956aac2b2d719e76c9c8cf6a77e9373021900307)

tirkarthi committed:
Fix crash when clearing run with task from normal to mapped (#31352)

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit f82246abe9491a49701abdb647be001d95db7e9f)

dstandish committed:
Add docstring and signature for _read_remote_logs (#31623)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit ce7766e0a52e15b2b1ef7e7f9c613ea686fbfca6)

hussein-awala committed:
Add an example for dynamic task mapping with non-TaskFlow operator (#29762)

(cherry picked from commit 4d4c2b9d8b5de4bf03524acf01a298c162e1d9e4)

r-richmond committed:
📝 Update release documentation (#31631)

(cherry picked from commit 54ac9234a8154325b13202e558c5fd8798319c7a)

vandonr-amz committed:
remove ducplicate entry in the metrics doc (#31654)

(cherry picked from commit fbb095605ab009869ef021535c16b62a3d18a562)

MSchutt committed:
Fix wrong cron syntax (#31666)

(cherry picked from commit 6a69fbb10c08f30c0cb22e2ba68f56f3a5d465aa)

amoghrajesh committed:
Wording the next and last run DAG columns better (#31467)

* Better wording the next and last run DAG columns

* review comments from brent

* review comments from Jed

* Undoing test changes

---------

Co-authored-by: Amogh <adesai@cloudera.com>
(cherry picked from commit 7db42fe6655c28330e80b8a062ef3e07968d6e76)

eladkal committed:
update path of KubernetesPodOperator in core docs (#31683)

(cherry picked from commit 8fcd2cf220382d8daed2e31fd3ed465e3ca5ee15)

ephraimbuddy committed:
Add back accidentally removed `create_session` in utils/db.py (#31705)

The create_session function was accidentally removed in commit
https://github.com/apache/airflow/commit/522661b6ad4479e3c8243b2d2c8a793d1af82c17
and it's a breaking change.

(cherry picked from commit 5f3a46427bd8fca6514b0080270f93600b033851)

potiuk committed:
Add severity rating explanation to our Security Policy (#31702)

(cherry picked from commit 2be5d3a78ee6bd5d15a18ca30ccfe51fbda2db87)

Adaverse committed:
Few doc fixes - links, grammer and wording (#31719)

* fix-docs

* Update BREEZE.rst

(cherry picked from commit 8dc895cbb64e991893705b04e03a6655c35888fe)

mdembiczak committed:
Fix `max_active_tis_per_dagrun` for Dynamic Task Mapping (#31406)

* assign filter output to variable

* add tests for max_active_tis_per_dagrun

* fix tests and code style

(cherry picked from commit 342b4fe836c864272ca298bb3dcac61f2eccac45)

potiuk committed:
Fix failing get_safe_url tests for latest Python 3.8 and 3.9 (#31766)

The latest release of Python 3.8 and 3.9 have been just released
that contain the fix to a security vulnerability backported to
those versions:

https://github.com/python/cpython/issues/102153

Release notes:
* https://www.python.org/downloads/release/python-3817/
* https://www.python.org/downloads/release/python-3917/

The fix improved sanitizing of the URLs and until Python 3.10 and
3.11 get released, we need to add the sanitization ourselves to
pass tests on all versions.

In order to improve security of airflow users and make the tests
work regardless whether the users have latest Python versions
released, we add extra sanitisation step to the URL to apply
the standard WHATWG specification.

(cherry picked from commit 87c5c9fa629317090ce65ec4c686596a2c4cd148)

vlazuka committed:
Include rediss to the list of supported URL schemes (#31028)

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 471fdacd853a5bcb190e1ffc017a4e650097ed69)

evanmiller committed:
Don't present an undefined execution date (#31196)

If a task had entered a Failed state but had not yet started, the UI
would display the current time as the "Run:" time. This is confusing.

(cherry picked from commit e16207409998b38b91c1f1697557d5c229ed32d1)

ephraimbuddy committed:
Fix typing for POST user endpoint (#31767)

* Fix typing for POST user endpoint

This should raise exception not return it.

* fixup! Fix typing for POST user endpoint

(cherry picked from commit f7ed87823afb2871d8f11e615d96693db3a6c2b8)

dstandish committed:
Replace unicodecsv with standard csv library (#31693)

unicodecsv appears to be missing a license which can cause trouble (see https://github.com/jdunck/python-unicodecsv/issues/80)

And it appears that this library may no longer be required.

(cherry picked from commit fbeb01cb17b7cb9c2e27ac7010f423a2bced78b4)

dstandish committed:
Control permissibility of driver config in extra from airflow.cfg (#31754)

Refines https://github.com/apache/airflow/pull/31713, which disabled (by default) setting driver through extra.  Here we make it so that the flag to enable is located in airflow config instead of hook param.

(cherry picked from commit 438ba41e142593f2b0916893eccbd08fbe4d277b)

hussein-awala committed:
Fix wrong update for nested group default args (#31776)

Signed-off-by: Hussein Awala <hussein@awala.fr>
(cherry picked from commit 8891c6944671d8965a63ddbd59b5c23597e86f0a)

jedcunningham committed:
Mark `[secrets] backend_kwargs` as a sensitive config (#31788)

(cherry picked from commit 8062756fa9e01eeeee1f2c6df74f376c0a526bd5)

Commit:
apply static checks

pankajastro committed:
Fix Kubernetes executor set wrong task status (#31274)

* Fix Kubernetes executor set wrong task status

In the case of multiple schedulers and lots of tasks running
If somehow schedulers restart and try to adopt pods
in some cases, it sets the wrong task status.
In this PR, I'm changing some checks so that if the pod status is non-terminal
then set the task status Failed only
if the pod event type is DELETED and POD_EXECUTOR_DONE_KEY is in the pod label

* cleanup

(cherry picked from commit dfbf5299ee2c3e68f6ec39ac502059fedbef3358)

Commit:
Update RELEASE_NOTES.rst

pierrejeambrun commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,94 @@
+- Few doc fixes - links, grammer and wording (#31719)
+- Add description in a few more places about adding airflow to pip install (#31448)
+
+Improvement Changes
```
I think semver requires a minor release for this, we can either mark them for 2.7.0 and drop them from the cherry picking, or consider them as 'misc/internal' if we want them in this patch. (if applicable)

eladkal replied:
All can be classified as misc I just thought they are a bit more important to note for the users

pierrejeambrun replied:
Yes, lets see what others think

eladkal replied:
if the issue is just about the title we can change it.. if the issue is about a specific PR that shouldn't be included in this release that is another issue :)

ephraimbuddy replied:
Most of the items in the improvement looks like real improvements and I suggest we move them to 2.7.0. I think we should be using patch releases only for bug fixes.

jedcunningham replied:
I could see reclassifying the 3.11 one as misc, but I think the other 2 should wait.

eladkal replied:
I removed/reclassified all the PRs that were under improvements.

pierrejeambrun commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,94 @@
+^^^^^^^^^^^^^^^^
+- update path of KubernetesPodOperator in core docs (#31683)
+- Fix wrong cron syntax (#31666)
+- 📝 Update release documentation (#31631)
```
Not sure how this icon will render on our website, mails and stuff. I would be for removing it

pierrejeambrun commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,94 @@
+- Executor events are not always "exited" here (#30859)
+- Rephrase comment in setup.py (#31312)
+
+Doc-only changes
```
```suggestion
Doc only changes
```

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,94 @@
+- Use keyword parameters for migration methods for mssql (#31309)
+- Control permissibility of driver config in extra from airflow.cfg (#31754)
+- Retain on_failure_fail_dagrun attr when other task attrs are overridden (#31678)
+- fixing broken links in openapi/v1.yaml (#31619)
```
```suggestion
- Fixing broken links in openapi/v1.yaml (#31619)
```

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,94 @@
+- Control permissibility of driver config in extra from airflow.cfg (#31754)
+- Retain on_failure_fail_dagrun attr when other task attrs are overridden (#31678)
+- fixing broken links in openapi/v1.yaml (#31619)
+- Hide old alert box when testing conn with different value (#31606)
```
```suggestion
- Hide old alert box when testing connection with different value (#31606)
```

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,94 @@
+- Resolving issue where Grid won't un-collapse when Details is collapsed (#31561)
+- Fix sorting of tags (#31553)
+- Add the missing ``map_index`` to the xcom key when skipping downstream tasks (#31541)
+- fix airflow users delete CLI command (#31539)
```
```suggestion
- Fix airflow users delete CLI command (#31539)
```

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,94 @@
+- Fix sorting of tags (#31553)
+- Add the missing ``map_index`` to the xcom key when skipping downstream tasks (#31541)
+- fix airflow users delete CLI command (#31539)
+- Include triggerer health status in Airflow /health endpoint (#31529)
```
```suggestion
- Include triggerer health status in Airflow `/health` endpoint (#31529)
```

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,94 @@
+- Update release documentation (#31631)
+- Add docstring and signature for _read_remote_logs (#31623)
+- Updating UI screenshots in docs (#31577)
+- Remove typo in docs serializers.rst (#31507)
```
```suggestion
```
I think we should remove these, doesn't seem important

ephraimbuddy replied:
```suggestion
```
I think we should remove these, doesn't seem important

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,94 @@
+- Fix version support information (#31468)
+- Add missing BashOperator import to documentation example (#31436)
+- Update the output of airflow info command in the doc (#31336)
+- Fix task.branch error caused by incorrect initial parameter (#31265)
```
```suggestion
```
This is breeze related. We can skip

eladkal replied:
it's not breeze but we can remove it if you'd like..
it's a doc only change

ephraimbuddy replied:
Sorry, it was the one below it that I was referring to, it has more of breeze change. Let's remove it

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,90 @@
+
+Bug Fixes
+^^^^^^^^^
+- Fix Kubernetes executor set wrong task status (#31274)
```
```suggestion
- Fix Kubernetes executors detection of deleted pods (#31274)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,90 @@
+- Fix Kubernetes executor set wrong task status (#31274)
+- Use keyword parameters for migration methods for mssql (#31309)
+- Control permissibility of driver config in extra from airflow.cfg (#31754)
+- Retain ``on_failure_fail_dagrun`` attr when other task attrs are overridden (#31678)
```
```suggestion
```

This should wait for 2.7, it's part of AIP-52.

eladkal replied:
removed

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,90 @@
+- Upgrade FAB to 4.3.1 (#31203)
+- Added metavar and choices to --state flag in airflow dags list-jobs CLI for suggesting valid state arguments. (#31308)
+- Use only one line for tmp dir log (#31170)
+- Executor events are not always "exited" here (#30859)
```
This feels like a bugfix to me.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,90 @@
+- Executor events are not always "exited" here (#30859)
+- Rephrase comment in setup.py (#31312)
+- Add fullname to owner on logging (#30185)
+- Validate connection IDs (#31140)
```
This also feels like a bugfix.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,90 @@
+- Add fullname to owner on logging (#30185)
+- Validate connection IDs (#31140)
+- Make connection id validation consistent across interface (#31282)
+- Support for sorting DAGs by Last Run Date in the web UI (#31234)
```
Was this brought in for cherrypicking ease? This feels out of place in a patch release.

eladkal replied:
Its a 2 line change. We can push it to 2.7.0
(In some casses we have a thin line between a misc change and a feature for example adding a templated_field)

eladkal replied:
pushed this one to 2.7.0

potiuk committed:
Bring back unicodecsv as depenendency of Airflow (#31814)

Removing unicodescv as dependency invites problems when users will
use older hive, google, microsoft providers, because they were
using unicodecsv, but they did not declare it as dependency (it
was a transitive dependency of the "apache-airflow" package).

It has been removed in #31693

Unicodecsv has very low footprint so this is not a problem to
keep it.

The dependency misses license in it's package, therefore we
add the licence in our "licences" folder.

(cherry picked from commit a853233ddb838873c7f0958402cccf5e25ea60c8)

eladkal committed:
Update RELEASE_NOTES.rst

potiuk committed:
Synchronize dev scripts with main

A number of dev scripts have been cherry-picked in a different order
from main and that made it quite a bit more difficult to apply all of
the cherry-picks consistently. Also we are supporting wider range of
Python versions (3.7 - 3.11) than either previous 2.6 or main,
so the breeze scripts and tests had to be updated to reflect
those differences.

Those changes are in dev tooling only, they have no impact on airflow
source code.

potiuk committed:
Remove yandex from 2.6 branch

potiuk committed:
Fix ERD diagram generated in 3.7 image

potiuk committed:
Fix selective checks skipping provider tests in non main branch (#31821)

When CI runs in non-main branch, some provider precommits and provider
tests should be disabled. There was a typo that prevented this from
happening, also names of the pre-commits changed.

(cherry picked from commit bc7e471cdae73a3c46ef89d17dbc1a54212b9291)

potiuk committed:
Update release notes

potiuk committed:
Fix spelling

dstandish committed:
Use single source of truth for sensitive config items (#31820)

Previously we had them defined both in constant and in config.yml.

Now just config.yml

(cherry picked from commit cab342ee010bfd048006ca458c760b37470b6ea5)

potiuk committed:
Fix failing AIP-52 test

potiuk committed:
Skip instalation of provider sdist packages when provider tests skipped (#31827)

The job was supposed to be skipped when provider tests were not run. But
it was not.

(cherry picked from commit 83ababc36894de662eb462a2f9e6bdf144b794a6)

potiuk committed:
Only run builds on "push" events for v* test branch (#31825)

We do not need to run separate build on push and pull request
when we are cherry-picking and pushing branch to apache/airflow
repo. Those PRs are always made by committers, and they are done
as direct push to the apache/airflow repo so they will run build
on push (which will be equivalent of canary run). We also do not
need to run anything on stable, because stable is the result of
merging those changes from test after they get green.

(cherry picked from commit 0dddaa0adc323bb49772ac9f53e4ba7668dbe522)

potiuk committed:
Check that breeze image builds quickly only on main branch (#31828)

This check is really for the developers who are developing on main
and there is no need to run it for other branches.

(cherry picked from commit da4912b5e562c7a30e0c54f79220c99a32e69ab9)

potiuk committed:
Update release notes

potiuk reviewed (approved):
LGTM

pierrejeambrun committed:
Update release note and revision head

pierrejeambrun committed:
Fix Update Airflow version to 2.6.2

potiuk committed:
Change working directory for breeze unit tests (#31839)

The recent version of pytest released on June 10th introduced
change that it will read conftest tests starting in the directory
which is current working directory, this causes breeze tests run
with working directory being top airflow directory to fail because
it's requirements do not contain a few libraries that are used
in airflow's conftest.py

This change rather than specifying directory at pytest command
line, changes working directory for the tests.

(cherry picked from commit 320671914ae2ebce30bb888ef2125bf79b1ba39e)
